flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1984,6 +1988,18 @@ IntrinsicLibrary::genChar(mlir::Type type,
   return fir::CharBoxValue{cast, len};
 }
 
+// COMMAND_ARGUMENT_COUNT
+fir::ExtendedValue
+IntrinsicLibrary::genCommandArgumentCount(mlir::Type resultType,
+                                          llvm::ArrayRef<fir::ExtendedValue> args) {
+  assert(args.size() == 0);
+  [[maybe_unused]] auto ty = builder.getIntegerType(builder.getKindMap().getIntegerBitsize(
+                                   builder.getKindMap().defaultIntegerKind()));
```

Comment:
Moving `builder.getIntegerType(builder.getKindMap().getIntegerBitsize(builder.getKindMap().defaultIntegerKind()))` in a `builder.getDefaultIntegerType()` builder member function could make the code clearer here (and might be useful in other places).

Reply:
Good idea! Done.